### PR TITLE
Add metadata container on ovn-standalone role

### PR DESCRIPTION
**Pull request by averdagu**

### CODE REVIEW

File: edpm_ansible/roles/edpm_ovn/templates/ovn_metadata_agent.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+image: "{{ edpm_ovn_metadata_agent_image }}"
+net: host
+privileged: true
+user: root
+restart: always
+depends_on:
+  - openvswitch.service
+healthcheck:
+  test: '/openstack/healthcheck {{ edpm_ovn_sb_server_port }}'
+volumes:
```
Comment (bshephar) [inline edpm_ansible/roles/edpm_ovn/templates/ovn_metadata_agent.yaml.j2:11]: I guess we also need to add the TLS volumes in cases where TLS is enabled?
```
edpm_ovn_controller_tls_volumes
```

Like: https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/edpm_ansible/roles/edpm_ovn/templates/ovn_controller.yaml.j2#L19-L24

Comment (bshephar) [reply edpm_ansible/roles/edpm_ovn/templates/ovn_metadata_agent.yaml.j2:11]: Oh right, those volumes are for ovn_controller, my bad. So metadata agent doesn't care about TLS ever?

Comment (bshephar) [reply edpm_ansible/roles/edpm_ovn/templates/ovn_metadata_agent.yaml.j2:11]: Ok, seems fine. Just looked into the old issue we had in TripleO related to TLS and the metadata agent:
https://review.opendev.org/c/openstack/tripleo-heat-templates/+/605406/3/puppet/services/ovn-metadata.yaml


But it seems just changing these two variables for TLS-e deployments would be all we need to do:
https://github.com/openstack-k8s-operators/edpm-ansible/pull/34/commits/c31541d6fa2e4650891be1a35f40241f7d0255b2#diff-61d75487470ab69a16e68b27db851584d6cb094c5a27e3dce29a9e8d2185ef4fR153-R154


All good, seems like we're covered for that scenario here then.

**Comment by rebtoor** [comment]
@averdagu hi there! Can you fix the linting errors? They're just EOF missing new lines.
```Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing edpm_ansible/roles/edpm_ovn/tasks/configure.yml
Fixing edpm_ansible/roles/edpm_ovn/defaults/main.yml```

**Comment by rebtoor** [comment]
recheck

**Comment by averdagu** [comment]
The error is:

`Couldn't resolve module/action 'config_template'. This often indicates a misspelling, missing collection, or incorrect module path`

I thought that this was fixed with this [change](https://github.com/openstack-k8s-operators/edpm-ansible/pull/34/commits/5f42e42cf0b1d9ef22495496d31d831c6680bce4#diff-119e4fe6a7bde648cb6b0765db65367015247ae792310d88588a2f6c1aff3308)

File: edpm_ansible/roles/edpm_ovn/tasks/configure.yml [hunk 2]
```diff
@@ -45,3 +45,82 @@
...
+        cp -a ${mount_dir}/etc/neutron/neutron_ovn_metadata_agent.ini {{ edpm_ovn_neutron_config_dir }}/etc/neutron/plugins/networking-ovn/networking-ovn-metadata-agent.ini
+      failed_when: false
+      register: copy_config_files
+
+    - name: Remove neutron_metadata_config container
+      ansible.builtin.shell: >
+        podman rm -f neutron_metadata_config || :
+
+    - name: Configure neutron.conf
+      config_template:
```
Comment (rebtoor) [inline edpm_ansible/roles/edpm_ovn/tasks/configure.yml:108]: please, replace the task with its fqcn: `openstack.config_template.config_template`

File: edpm_ansible/roles/edpm_ovn/tasks/configure.yml [hunk 3]
```diff
@@ -45,3 +45,82 @@
...
+        src: "{{ edpm_ovn_neutron_config_dir }}/etc/neutron/neutron.conf"
+        dest: "{{ edpm_ovn_neutron_config_dir }}/etc/neutron/neutron.conf"
+        config_type: ini
+        config_overrides: "{{ edpm_ovn_metadata_agent_neutron_config | combine(edpm_ovn_metadata_agent_neutron_config_overrides, recursive=True) }}"
+
+    - name: Configure rootwrap.conf
+      config_template:
```
Comment (rebtoor) [inline edpm_ansible/roles/edpm_ovn/tasks/configure.yml:115]: please, replace the task with its fqcn: `openstack.config_template.config_template`

File: edpm_ansible/roles/edpm_ovn/molecule/default/prepare.yml [hunk 4]
```diff
@@ -26,6 +26,7 @@
         test_deps_setup_stream: true
         test_deps_edpm_packages:
           - openvswitch
+          - ansible-config_template
```
Comment (rebtoor) [inline edpm_ansible/roles/edpm_ovn/molecule/default/prepare.yml:29]: we don't use the package, we import the collection in `requirements.yml`

**Review by rebtoor — CHANGES_REQUESTED** [review]

File: edpm_ansible/roles/edpm_ovn/molecule/default/prepare.yml (hunk 4 above)
Comment (averdagu) [reply edpm_ansible/roles/edpm_ovn/molecule/default/prepare.yml]: Didn't know that, thanks!

File: edpm_ansible/roles/edpm_ovn/tasks/configure.yml [hunk 5]
```diff
@@ -45,3 +45,82 @@
...
+        src: "{{ edpm_ovn_neutron_config_dir }}/etc/neutron/rootwrap.conf"
+        dest: "{{ edpm_ovn_neutron_config_dir }}/etc/neutron/rootwrap.conf"
+        config_type: ini
+        config_overrides: "{{ edpm_ovn_metadata_agent_rootwrap_config | combine(edpm_ovn_metadata_agent_rootwrap_config_overrides, recursive=True) }}"
+
+    - name: Configure networking-ovn-metadata-agent.ini
+      config_template:
```
[the remark this answers is not in the document]
Comment (averdagu) [reply edpm_ansible/roles/edpm_ovn/tasks/configure.yml]: Thanks, done

File: edpm_ansible/roles/edpm_ovn/tasks/configure.yml (hunk 2 above)
Comment (averdagu) [reply edpm_ansible/roles/edpm_ovn/tasks/configure.yml]: Thanks, done

**Comment by rebtoor** [comment]
recheck

**Review by rebtoor — APPROVED** [review]

**Pull request merged**